includes/Main.php

```diff
@@ -1135,6 +1135,46 @@ public function schedule_temporary_files_cleanup() {
 		$delete_timestamp = time() - ( intval ( DAY_IN_SECONDS * $cleanup_age_days ) );
 		$this->temporary_files_cleanup( $delete_timestamp );
 	}
+	
+	/**
+	 * Recursive function to list files up to a certain folder depth
+	 *
+	 * @param string $path
+	 * @param int    $depth
+	 * @param int    $max_depth
+	 * @param array  $excluded_files
+	 *
+	 * @return array
+	 */
+	private function list_files_with_depth( string $path, int $depth, int $max_depth, array $excluded_files = array() ): array {
```

Comment:
May I ask why this method has been introduced to be replaced with `list_files()`?

Reply:
Since I needed to replace `wp_delete_file` as you noticed, we could also use `WP_Filesystem` to replace `list_files`. The only limitation is the recursion depth, but we might be able to overlook that.